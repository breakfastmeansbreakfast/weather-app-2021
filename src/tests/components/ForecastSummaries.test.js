import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1525392000000,
      description: "Stub description 1",
      icon: "800",
      temperature: {
        max: 22,
        min: 12,
      },
    },
    {
      date: 1524382000000,
      description: "Stub description2",
      icon: "200",
      temperature: {
        max: 24,
        min: 13,
      },
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps} />
    );

    expect(getAllByTestId("date-id")).toHaveLength(2);
    expect(getAllByTestId("temperature-id")).toHaveLength(2);
    expect(getAllByTestId("description-id")).toHaveLength(2);
    expect(getAllByTestId("icon-image-id")).toHaveLength(2);
  });
});
