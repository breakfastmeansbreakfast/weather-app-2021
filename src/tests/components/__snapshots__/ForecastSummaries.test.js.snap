// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
        data-testid="date-id"
      >
        4 May 2018
      </div>
      <div
        class="forecast-summary__temperature"
        data-testid="temperature-id"
      >
        22°C
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="icon-image-id"
      >
        <i
          class="wi wi-day-sunny wi-flip-horizontal wi-rotate-90"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__description"
        data-testid="description-id"
      >
        Stub description 1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
        data-testid="date-id"
      >
        22 April 2018
      </div>
      <div
        class="forecast-summary__temperature"
        data-testid="temperature-id"
      >
        24°C
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="icon-image-id"
      >
        <i
          class="wi wi-day-storm-showers wi-flip-horizontal wi-rotate-90"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__description"
        data-testid="description-id"
      >
        Stub description2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
